core/src/main/scala/org/apache/spark/ErrorInfo.scala

```diff
@@ -71,11 +71,18 @@ private[spark] object SparkThrowableHelper {
     mapper.readValue(errorClassesUrl, new TypeReference[SortedMap[String, ErrorInfo]]() {})
   }
 
+  def getMessage(
+      errorClass: String,
+      errorSubClass: String,
+      messageParameters: Array[String]): String = {
+    getMessage(errorClass, errorSubClass, messageParameters, None)
+  }
+
   def getMessage(
       errorClass: String,
       errorSubClass: String,
       messageParameters: Array[String],
-      queryContext: String = ""): String = {
+      queryContext: Option[QueryContext]): String = {
```

Comment:
I think we just need a string parameter here.

Reply:
I created an interface which extends `QueryContext` and has an additional method to get textual summary.